recipes/equilibrator-cache/meta.yaml

```
@@ -0,0 +1,54 @@
+{% set name = "equilibrator-cache" %}
+{% set version = "0.4.0b2" %}
+
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/equilibrator-cache-{{ version }}.tar.gz
+  sha256: 83e33d3505f3f1557863fb6ae606017181c06a5451e7b842b41bb3ef71d639e5
+
+build:
+  number: 0
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv --no-deps
+
+requirements:
+  host:
+    - pip
+    - python >=3.6
+  run:
+    - appdirs ~=1.4
+    - httpx ~=0.16
+    - importlib_resources
+    - numpy ~=1.19
+    - pandas ~=1.0
+    - periodictable ~=1.5
+    - pint ~=0.14
+    - python >=3.6
+    - python-levenshtein
+    - pyzenodo3 ~=1.0
+    - scipy ~=1.5
+    - sqlalchemy ~=1.3
+    - tenacity ~=6.2
+    - tqdm ~=4.50
+
+test:
+  imports:
+    - equilibrator_cache
+    - equilibrator_cache.models
+  requires:
+    - pip
```

I don't think you need pip to run the tests, so you can remove the `requires` section.

---

As there are a lot of tight version pins I'd be inclined to add this to make sure the version metadata doesn't get out of sync:
```suggestion
    - pip
  commands:
    - pip check
```

---

I removed `pip check` because the levenshtein package has different names for the normal Python package and the conda package which caused `pip check` to complain.

---

The Python metadata should be correct regardless of the conda-forge package name. Do you have a CI log I can take a look at?

---

Just added this in you can inspect the error logs.

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=236226&view=logs&j=10bda42a-d420-5c28-a43f-bc4e66405e5c&t=bef775d0-a1a2-55a3-9721-17147029baf0&l=796

---

Ah because:
* `equilibrator-cache` depends on `python-Levenshtein-wheels` not `python-Levenshtein`
* `python-Levenshtein-wheels` fork of `python-Levenshtein` that simply adds wheels to PyPI

My suggestion would be to use the patch I've just committed (3aa9f07) to change the setuptools metadata and avoid the wheels mess. If you change the version constraints frequently the patch might fail to apply, if so you could rearrange the list of dependencies upstream and add whitespace to make the `setup.cfg` file more amiable to being patched. (This is also the situation in which pip-check will be most useful.)

@midnighter as it looks like you're a developer of `equilibrator-cache` so you probably have a better idea of what the situation actually is. I'm happy to merge this regardless of what you want to do so feel free to revert my changes.

---

That's a neat solution, thanks! We will double check whether we can do without the wheels package and then proceed accordingly.